src/Citeproc/Eval.hs

```diff
@@ -1142,16 +1148,64 @@ evalItem layout (position, item) = do
            , contextPosition = position
            },
         st{ stateReference = ref
-          , stateUsedYearSuffix = False }))
+          , stateUsedYearSuffix = False
+          , stateUsedIdentifier = False
+          , stateUsedTitle = False
+          }))
         $ do xs <- mconcat <$> mapM eElement (layoutElements layout)
+
+             -- find identifiers that can be used to hyperlink the title 
+             let mbident = 
+                    foldl (<|>) Nothing
+                      [ IdentDOI   <$> (valToText =<< lookupVariable "DOI" ref)
+                      , IdentPMCID <$> (valToText =<< lookupVariable "PMCID" ref)
+                      , IdentPMID  <$> (valToText =<< lookupVariable "PMID" ref)
+                      , IdentURL   <$> (valToText =<< lookupVariable "URL" ref)
+                      ]
+             let mburl = identifierToURL <$> mbident
+            
+             -- hyperlink any titles in the output
+             let linkTitle url (Tagged TagTitle x) = Linked LinkTitle url [x]
+                 linkTitle _ x = x
```

Comment:
Here's a thought:  instead of `Linked LinkTitle url [x]`, what about `Linked url [Tagged TagTitle x)`?  The extra parameter on `Linked` wouldn't really be needed, since we could just check to see if the content is a title.

Reply:
Seems reasonable, so I made the change.